app/partials/settings/security.jade

```diff
@@ -46,7 +46,8 @@ h5.well.type-h5.em-400.hidden-xs(translate="SECURITY_BASIC_EXPLAIN")
         p.em-300.text.hidden-xs(translate="WALLET_SECOND_PASSWORD_EXPLAIN" ng-hide="userHasExchangeAcct")    
         p.em-300.text.hidden-xs(translate="WALLET_SECOND_PASSWORD_DISABLED_EXPLAIN" ng-show="userHasExchangeAcct")
       .col-sm-12.col-md-6.setting-result(ng-hide="active || removing")
-        button.button-primary.display-inline-block(ng-click="activate()", translate="SET_SECOND_PASSWORD", ng-show="!settings.secondPassword" ng-disabled="userHasExchangeAcct")
+        button.button-primary.display-inline-block(ng-click="activate()", translate="SET_SECOND_PASSWORD", ng-show="showButton()" ng-hide="hideButton()" ng-disabled="userHasExchangeAcct")
+        button.button-primary.display-inline-block(ng-click="promptRecovery()", translate="SET_SECOND_PASSWORD", ng-hide="walletStatus.didConfirmRecoveryPhrase || settings.secondPassword || walletStatus.dismissedRecoveryPrompt" ng-show="!settings.secondPassword && !walletStatus.didConfirmRecoveryPhrase" ng-disabled="userHasExchangeAcct")
```

Comment:
Can these be the same element? You might be able to do something like `ng-click="walletStatus.didConfirmRecoveryPhrase ? activate() : promptRecovery()`. That would simplify the show / hide logic as well.